src/System.Buffers.Experimental/System/Buffers/OwnedArray.cs

```diff
@@ -44,7 +41,7 @@ public override MemoryHandle Pin(int byteOffset = 0)
                 if (byteOffset != 0 && (((uint)byteOffset) - 1) / Unsafe.SizeOf<T>() >= _array.Length) throw new ArgumentOutOfRangeException(nameof(byteOffset));
```

Comment:
This check should just be `if ((uint)elementIndex > (uint)_array.Length) { FAIL; }`.
(By convention we allow elementIndex = _array.Length to not fail.)